### PR TITLE
Feat/did precompile

### PR DESCRIPTION
**Pull request by niall-watr**

### CODE REVIEW

File: precompiles/did/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+#[precompile_utils::generate_function_selector]
+#[derive(Debug, PartialEq)]
+pub enum Action {
+	CreateDID = "create_did(address, address, uint8[], bytes[])",
```
Comment (Moliholy) [inline precompiles/did/src/lib.rs:27]: According to [Solidity's syntax guidelines](https://docs.soliditylang.org/en/v0.8.19/style-guide.html#function-names) it's recommended to use camel case notation to name functions. I've actually seen Moonbean's example for XC-20 tokens and they use the aforementioned style.

Comment (Moliholy) [inline precompiles/did/src/lib.rs:27]: Also, function selector must not contain spaces. A valid function selector would be this one:

```
createDid(address,address,uint8[],bytes[])
```

Comment (niall-watr) [reply precompiles/did/src/lib.rs]: thanks Jose, keep the comments coming!

**Comment by Moliholy** [comment]
Update to the latest `main`.

**Review by peterwht — APPROVED** [review]

**Pull request merged**